### DISCUSSION TITLE
UnicodeEncodeError: 'locale' codec can't encode character '\u65f6' in position 2: encoding error

### DISCUSSION THREAD

**Discussion by qq995002966** [pyinstaller/pyinstaller#7997] [Help] [answered]
- python version : 3.10_x64
- pyinstaller version : 6.0.0
- windows version : Windows 10 Enterprise 1000.19041.1000.0

I writen a small tool using python, it works well in pycharm. However after bundle by PyInstaller it will crash with a encoding error:
```
Traceback (most recent call last):
  File "login.py", line 124, in <module>
  File "login.py", line 31, in __init__
  File "login.py", line 96, in load_config
  File "ui\WindowBase.py", line 27, in log
UnicodeEncodeError: 'locale' codec can't encode character '\u65f6' in position 2: encoding error
```
Bellow is my code in the stack trace. All files encoding in my project is 'utf-8'. Any help would be appreciated.

![2](https://github.com/pyinstaller/pyinstaller/assets/7163238/38ac1967-aa3d-4412-ae83-fe024817d653)
![3](https://github.com/pyinstaller/pyinstaller/assets/7163238/fe57fa9b-5fcb-40c4-9c64-15729ef76803)

**Comment by rokm** [comment, marked as answer]
Trying to put `\u65f6` (`时`) into strftime's `format` string seems to raise the same error when I try to run the program in python (3.10.8 64-bit Windows installer from python.org), no PyInstaller involved:
```python
# program.py
import datetime
print(datetime.datetime.now().strftime('%H时%S'))
```

```
Traceback (most recent call last):
  File "[...]\program.py", line 3, in <module>
    print(datetime.datetime.now().strftime('%H时%S'))
UnicodeEncodeError: 'locale' codec can't encode character '\u65f6' in position 2: encoding error
```

From what I understand, `datetime.strftime()` is calling into CRT functions, so behavior might be dependent on environment's locale settings and thus not portable anyway.

I'd suggest you apply a replacement work-around (use format string with ascii characters only, and replace the delimiters with desired unicode characters in postprocessing), or use the encode/decode with unicode escapes from https://github.com/sphinx-doc/sphinx/blob/8ae43b9fd4b68473d25e872766a8786a5be85852/sphinx/util/osutil.py#L172-L176, e.g.:


```python
# program.py
import datetime
print(datetime.datetime.now().strftime('%H时%S'.encode('unicode-escape').decode()).encode().decode('unicode-escape'))
```

**Reply by qq995002966** [reply]
thanks alot. This help me great. 👍
